src/url-loader/prefixed-url-loader.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * @license
+ * Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {UrlLoader} from './url-loader';
+
+/**
+ * Resolves requests via a given delegate loader for URLs matching a given
+ * prefix.  URLs are provided to their delegate without the prefix.
+ */
+export class PrefixedUrlLoader implements UrlLoader {
+  constructor(public prefix: string, public delegate: UrlLoader) {
+  }
+
+  canLoad(url: string): boolean {
+    return url.startsWith(this.prefix) &&
+        this.delegate.canLoad(this.unprefix(url));
+  }
+
+  async load(url: string): Promise<string> {
+    if (!url.startsWith(this.prefix)) {
```

Comment:
OH WAIT - you were saying the load shouldn't actually get called with a bad URL.  Yeah most of the time that's right, because canLoad would be called in advance of load to verify.

So yeah you're right about that.  Shouldn't happen-- but if it did, this guard ensures it doesn't send the load request to the delegate.